lib/tlog/sink.c

```diff
@@ -135,6 +135,20 @@ tlog_sink_flush(struct tlog_sink *sink)
     return grc;
 }
 
+void
+tlog_sink_io_close(struct tlog_sink *sink,
+                bool output)
+{
+    assert(sink == NULL || tlog_sink_is_valid(sink));
+
+    if (sink == NULL) {
+        return;
+    }
```

Comment:
I think the sink cannot be NULL if we want to operate on it. This is not like we're destroying it so we don't care it's already destroyed. So we shouldn't be trying to handle it being NULL, and should do a hard abort when it happens. I.e. remove `sink == NULL` from the `assert` and drop that return condition.